core/deprecation_manager.h

```diff
@@ -0,0 +1,72 @@
+#include "core/hash_map.h"
+#include "core/ustring.h"
+
+#ifndef DEPRECATION_MANAGER_H
+#define DEPRECATION_MANAGER_H
+
+#ifdef TOOLS_ENABLED
```

Comment:
Should be `DEBUG_METHOD_ENABLED`.

Reply:
I would just use DEBUG_ENABLED here then. Maybe DEBUG_METHODS_ENABLED should be passed to the compiler like other macros.

Reply:
Fixed